src/compiler/utilities.ts

```diff
@@ -3294,11 +3294,21 @@ namespace ts {
         return !!name && isDynamicName(name);
     }
 
+    export function getComputedPropertyNameExpression(name: ComputedPropertyName): Expression {
+        return isElementAccessExpression(name.expression)
+                    ? skipParentheses(name.expression.argumentExpression)
+                    : name.expression;
+    }
+
     export function isDynamicName(name: DeclarationName): boolean {
         if (!(name.kind === SyntaxKind.ComputedPropertyName || name.kind === SyntaxKind.ElementAccessExpression)) {
             return false;
         }
-        const expr = isElementAccessExpression(name) ? skipParentheses(name.argumentExpression) : name.expression;
+
+        const expr = isElementAccessExpression(name)
+        ? skipParentheses(name.argumentExpression)
+        : getComputedPropertyNameExpression(name);

```

Comment:
I think this should just be `const expr = getComputedPropertyNameExpression(name)`; this conditional expression is just repeating what `getComputedPropertyNameExpression` does in a different order.

Reply:
it is repetitive, but it seems extracting the expression from `name` does differ a bit as we need to handle both `ElementAccessExpression | ComputedPropertyName`

for just `ElementAccessExpression`,  `name.argumentExpression` is used

and for `ComputedPropertyName`, both `name.expression` and `name.expression.argumentExpression` are used

i think it's ok?

Reply:
Ah, you're right.

Reply:
Then, seems okay, though the formatting here is quite odd.

Reply:
do you mean it should be one line:
```typescript 
const expr = isElementAccessExpression(name) ? skipParentheses(name.argumentExpression) : getComputedPropertyNameExpression(name)
```

or something else?

Reply:
That or have the lines underneath indented.